lib/authentication/auth_default.js

```diff
@@ -10,19 +10,53 @@
  * @returns {Object}
  * @constructor
  */
-function AuthDefault(password) {
+function AuthDefault(connectionConfig) {
+  const password = connectionConfig.password;
+  const mfaToken = connectionConfig.mfaToken;
+  const passcode = connectionConfig.getPasscode();
+  const isPasscodeInPassword = connectionConfig.getPasscodeInPassword();
+  
   /**
-   * Update JSON body with password.
-   *
-   * @param {JSON} body
-   *
-   * @returns {null}
-   */
+     * Update JSON body with password or token.
+     *
+     * @param {JSON} body
+     *
+     * @returns {null}
+     */
   this.updateBody = function (body) {
     body['data']['PASSWORD'] = password;
+
+    if (isMFAAuth()) {
+      setMFASessionParams(body);
+    }
   };
 
-  this.authenticate = async function () {};
+  function isMFAAuth() {
+    return ( connectionConfig.getAuthenticator() === 'USERNAME_PASSWORD_MFA' || mfaToken || passcode || isPasscodeInPassword);
```

Comment:
Please use authenticationTypes.MFA_TOKEN_AUTHENTICATOR instead of 'USERNAME_PASSWORD_MFA'